src/main/java/org/openstreetmap/atlas/utilities/caching/strategies/GlobalNamespaceCachingStrategy.java

```diff
@@ -0,0 +1,65 @@
+package org.openstreetmap.atlas.utilities.caching.strategies;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.function.Function;
+
+import org.openstreetmap.atlas.streaming.resource.Resource;
+
+/**
+ * A special case of {@link NamespaceCachingStrategy} that uses a predefined, global namespace. This
+ * means that all instances of {@link GlobalNamespaceCachingStrategy} will share the same underlying
+ * contents. From this it follows that fetches and invalidates will manifest across instances. To
+ * prevent concurrency issues, the global namespace is locked using a class lock. It is worth noting
+ * that this still does not protect the namespace's integrity from multiple JVMs running concurrent
+ * {@link GlobalNamespaceCachingStrategy} objects.
+ *
+ * @author lcram
+ */
+public class GlobalNamespaceCachingStrategy extends NamespaceCachingStrategy
+{
+    /*
+     * This is a random SHA256 hash. Collisions with this namespace are astronomically unlikely (due
+     * to the 256 bits of entropy).
+     */
+    private static final String GLOBAL_NAMESPACE = "3707740A818531237051A0F1E086CF701E2C38483675FCD1AAD8F5C5C33F19BC";
```

Comment:
In the Spark use-case, we would want to not do that, right? So multiple JVMs on the same node can still share the same cache?

Reply:
Well, it's a tradeoff worth discussing. Multiple JVMs sharing the same underlying cache would be good for performance (since it avoids downloading duplicate `Resources`). However, once you allow that you have no way to guarantee that a bad interleaving of `invalidate(my_uri)` in one JVM and `get(my_uri)` in another JVM won't occur, since at that point you are at the mercy of the OS scheduler.

Reply:
IMO the name "global" should apply to as a large a space as is reasonably expectable, in this case system-wide. If we want a JVM specific cache, then that's worth either a toggle in the `GlobalNamespaceCachingStrategy` constructor or another subclass explicitly dedicated to that.

Reply:
Fair enough - that makes sense. I really don't like the intrinsically unsafe nature of this code, but I am not sure how we could make it safe given the design constraints.